app/components/doi-funding-reference.js

```diff
@@ -48,18 +48,28 @@ export default Component.extend({
     }
   },
   updateFunderReference(funder) {
-    if (funder.uri) {
-      this.fragment.set('funderName', funder.name);
-      this.fragment.set('funderIdentifierType', 'Crossref Funder ID');
-      this.fragment.set('schemeUri', 'https://www.crossref.org/services/funder-registry/');
-      this.fragment.set('funderIdentifier', funder.uri);
-      this.set('isCrossrefId', true);
-    } else {
-      this.fragment.set('funderIdentifierType', 'Other');
-      this.fragment.set('funderIdentifier', null);
-      this.fragment.set('funderName', funder);
-      this.updateFunderSchemeAndType(null);
-      this.set('isCrossrefId', false);
+    switch (true) {
+      case funder === null:
```

Comment:
I'm not saying this is wrong, but javascript "falsey" can be tricky at times, is it definitely null?

Reply:
definetely null.  When the users click on the `X`  in the selector it sends `null` and then we clear all the fields.